### ISSUE TITLE
Error on Grails 3.1.x with Mongo and new Elasticsearch plugin

### ISSUE THREAD

**Issue by hugheba** [noamt/elasticsearch-grails-plugin#145]
I'm having an error when using the Mongo Grails plugin v5.0.1 with the Elasticsearch plugin v1.0.0.

I've created a simple sample project that recreates the error below.

[https://github.com/hugheba/grails3.1.x-mongo-elasticsearch](https://github.com/hugheba/grails3.1.x-mongo-elasticsearch)

```
ERROR org.springframework.boot.SpringApplication - Application startup failed
org.grails.core.exceptions.GrailsConfigurationException: Error configuring dynamic methods for plugin [elasticsearch:1.0.0]: Cannot add new static method [search] for arguments [[class java.lang.String]]. It already exists!
	at grails.plugins.DefaultGrailsPluginManager.doDynamicMethods(DefaultGrailsPluginManager.java:714) ~[grails-core-3.1.2.jar:3.1.2]
	at grails.boot.config.GrailsApplicationPostProcessor.onApplicationEvent(GrailsApplicationPostProcessor.groovy:227) ~[grails-core-3.1.2.jar:3.1.2]
	at grails.boot.config.GrailsApplicationPostProcessor.onApplicationEvent(GrailsApplicationPostProcessor.groovy) ~[grails-core-3.1.2.jar:3.1.2]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:163) ~[spring-context-4.2.4.RELEASE.jar:4.2.4.RELEASE]
	at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:136) ~[spring-context-4.2.4.RELEASE.jar:4.2.4.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:381) ~[spring-context-4.2.4.RELEASE.jar:4.2.4.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:335) ~[spring-context-4.2.4.RELEASE.jar:4.2.4.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.finishRefresh(AbstractApplicationContext.java:855) ~[spring-context-4.2.4.RELEASE.jar:4.2.4.RELEASE]
	at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.finishRefresh(EmbeddedWebApplicationContext.java:140) ~[spring-boot-1.3.2.RELEASE.jar:1.3.2.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:541) ~[spring-context-4.2.4.RELEASE.jar:4.2.4.RELEASE]
	at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.refresh(EmbeddedWebApplicationContext.java:118) ~[spring-boot-1.3.2.RELEASE.jar:1.3.2.RELEASE]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:766) ~[spring-boot-1.3.2.RELEASE.jar:1.3.2.RELEASE]
	at org.springframework.boot.SpringApplication.createAndRefreshContext(SpringApplication.java:361) ~[spring-boot-1.3.2.RELEASE.jar:1.3.2.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:307) ~[spring-boot-1.3.2.RELEASE.jar:1.3.2.RELEASE]
	at grails.boot.GrailsApp.run(GrailsApp.groovy:55) [grails-core-3.1.2.jar:3.1.2]
	at grails.boot.GrailsApp.run(GrailsApp.groovy:365) [grails-core-3.1.2.jar:3.1.2]
	at grails.boot.GrailsApp.run(GrailsApp.groovy:354) [grails-core-3.1.2.jar:3.1.2]
	at grails.boot.GrailsApp$run.call(Unknown Source) [grails-core-3.1.2.jar:3.1.2]
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48) [groovy-2.4.6.jar:2.4.6]
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113) [groovy-2.4.6.jar:2.4.6]
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:133) [groovy-2.4.6.jar:2.4.6]
	at g3.Application.main(Application.groovy:8) [main/:na]
Caused by: groovy.lang.GroovyRuntimeException: Cannot add new static method [search] for arguments [[class java.lang.String]]. It already exists!
	at groovy.lang.ExpandoMetaClass$ExpandoMetaProperty.registerStatic(ExpandoMetaClass.java:623) ~[groovy-2.4.6.jar:2.4.6]
	at groovy.lang.ExpandoMetaClass$ExpandoMetaProperty.registerIfClosure(ExpandoMetaClass.java:612) ~[groovy-2.4.6.jar:2.4.6]
	at groovy.lang.ExpandoMetaClass$ExpandoMetaProperty.leftShift(ExpandoMetaClass.java:596) ~[groovy-2.4.6.jar:2.4.6]
	at sun.reflect.GeneratedMethodAccessor66.invoke(Unknown Source) ~[na:na]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_74]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_74]
	at org.springsource.loaded.ri.ReflectiveInterceptor.jlrMethodInvoke(ReflectiveInterceptor.java:1426) ~[springloaded-1.2.5.RELEASE.jar:1.2.5.RELEASE]
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:210) ~[groovy-2.4.6.jar:2.4.6]
	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.call(PogoMetaMethodSite.java:71) ~[groovy-2.4.6.jar:2.4.6]
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48) [groovy-2.4.6.jar:2.4.6]
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113) [groovy-2.4.6.jar:2.4.6]
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125) [groovy-2.4.6.jar:2.4.6]
	at grails.plugins.elasticsearch.util.DomainDynamicMethodsUtils.injectDynamicMethods(DomainDynamicMethodsUtils.groovy:58) ~[elasticsearch-1.0.0.jar:na]
	at grails.plugins.elasticsearch.util.DomainDynamicMethodsUtils$injectDynamicMethods.call(Unknown Source) ~[na:na]
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48) [groovy-2.4.6.jar:2.4.6]
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:113) [groovy-2.4.6.jar:2.4.6]
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:133) [groovy-2.4.6.jar:2.4.6]
	at ElasticsearchGrailsPlugin.doWithDynamicMethods(ElasticsearchGrailsPlugin.groovy:150) ~[elasticsearch-1.0.0.jar:na]
	at org.grails.plugins.DefaultGrailsPlugin.doWithDynamicMethods(DefaultGrailsPlugin.java:750) ~[grails-core-3.1.2.jar:3.1.2]
	at grails.plugins.DefaultGrailsPluginManager.doDynamicMethods(DefaultGrailsPluginManager.java:711) ~[grails-core-3.1.2.jar:3.1.2]
	... 21 common frames omitted
```

**Comment by puneetbehl** [comment]
@hugheba This because the conflict in search method of MongoDB plugin and another we are trying to inject via Elasticsearch. I need to think through this because if we change the search method name, migrating to new will stop backward compatibility.

**Comment by hugheba** [comment]
@puneetbehl Would it be possible as a temporary solution to have a config option to not add the search method to Domain/Entities and use the elasticsearchSearch.service instead?

**Comment by puneetbehl** [comment]
@hugheba  Fixed in `v1.0.0.2` in commit  [7f6f1b0](https://github.com/noamt/elasticsearch-grails-plugin/commit/56e4204f50ea684b51cc4f5c8af23db6561f6d7d). 
Please read more about it under [dynamic-method-injection](http://noamt.github.io/elasticsearch-grails-plugin/docs/index.html#dynamic-method-injection) in [Documentation](http://noamt.github.io/elasticsearch-grails-plugin/docs/index.html)

**Comment by marcoscarceles** [comment]
@puneetbehl @hugheba The search method name has long been configurable:

http://noamt.github.io/elasticsearch-grails-plugin/docs/index.html#configuration

```
I'm using the latest ES plugin with the latest Mongo
